js/main.js

```diff
@@ -17,20 +17,55 @@ function formEvent(e) {
     notes: $form.elements.notes.value,
     entryId: data.nextEntryId
   };
-  data.nextEntryId++;
+  if (data.editing === null) {
+    data.nextEntryId++;
 
-  data.entries.unshift(formObj);
+    data.entries.unshift(formObj);
+    $img.setAttribute('src', 'images/placeholder-image-square.jpg');
+
+    var newDomTree = renderEntry(formObj);
+    $uldata.prepend(newDomTree);
+  }
+
+  if (data.editing !== null) {
+    var editedObj = {
+      entryId: data.editing.entryId,
+      notes: $form.elements.notes.value,
+      photo: $form.elements.photo.value,
+      title: $form.elements.title.value
+
+    };
+
+    for (var i = 0; i < data.entries.length; i++) {
+
+      if (data.entries[i].entryId === data.editing.entryId) {
+
+        data.entries.splice(i, 1, editedObj);
+      }
+      var $liEntries = document.querySelectorAll('li');
+
+      for (var a = 0; a < $liEntries.length; a++) {
+
+        var numLi = Number($liEntries[a].getAttribute('data-entry-id'));
+        if (numLi === editedObj.entryId) {
+          var newRender = renderEntry(editedObj);
+          $liEntries[a].replaceWith(newRender);
+          $header.textContent = 'New Entry';
+        }
+      }
```

Comment:
right now, you are doing this entire loop on every iteration of the outer loop 🤢 

You should just do it in the conditional for when you found the right entry above this.